lib/better_html/test_helper/safe_erb/tag_interpolation.rb

```diff
@@ -151,18 +151,29 @@ def validate_tag_interpolation(parent_node, ruby_node, attr_name)
         end
 
         def no_unsafe_calls(parent_node, ruby_node)
+          raw_error = false
+          to_json_input_to_raw = false
+          
           ruby_node.descendants(:send, :csend).each do |call|
             if call.method_name?(:raw)
-              add_error(
-                "erb interpolation with '<%= raw(...) %>' in this context is never safe",
-                location: nested_location(parent_node, ruby_node)
-              )
+              raw_error = true
             elsif call.method_name?(:html_safe)
               add_error(
-                "erb interpolation with '<%= (...).html_safe %>' in this context is never safe",
+                "erb interpolation with '<%= (...).html_safe %>' in this context may introduce XSS. Please request security review.",
                 location: nested_location(parent_node, ruby_node)
               )
             end
+
+            if raw_error && call.method_name?(:to_json)
+              to_json_input_to_raw = true
+            end
```

Comment:
the fact that we find .to_json in a subsequent iteration to the raw call means that it is higher up the call stack. If to_json is called _after_ raw - e.g. `raw(something).to_json` - then this will not trigger.